### PR TITLE
Fixes to be compatible with python3 or newer versions of torch

### PR DESCRIPTION
**Pull request by hyonaldo**
There is an issue that was not compatible with python3 or newer versions of torch.

For example, the issue below is some of those problems.
image

Thus I propose this pull request to fix those problems.
I have confirmed in the below environment that all code works perfectly well.
image

### CODE REVIEW

**Comment by jrzaurin** [comment]
I can't see it clearly in notebook, could you update the pull request to be applied on the .py files? 

This way I can see if clearly and merge

**Comment by hyonaldo** [comment]
@jrzaurin 
Thank you for your review.
There are ".py" file changes already. :)
e.g. https://github.com/jrzaurin/Wide-and-Deep-PyTorch/pull/2/files#diff-586c210e7820aab33c52dde1ef555b22

However, I think ".ipynb" files also require an update to work well on python3. So I have got rid of output from ".ipynb" notebooks to be able to see the changes only.

Not only that, I have confirmed that this PR version works well in both environments python2 and python3.

File: wide_deep/torch_model.py [hunk 1]
```diff
@@ -107,7 +106,7 @@ def compile(self, method="logistic", optimizer="Adam", learning_rate=0.001, mome
         if method == 'regression':
             self.activation, self.criterion = None, F.mse_loss
         if method == 'logistic':
-            self.activation, self.criterion = F.sigmoid, F.binary_cross_entropy
+            self.activation, self.criterion = torch.sigmoid, F.binary_cross_entropy
```
Comment (jrzaurin) [inline wide_deep/torch_model.py]: why `torch.sigmoid`? the functional `F.sigmoid` is actually more efficient?

File: wide_deep/torch_model.py [hunk 2]
```diff
@@ -191,8 +192,12 @@ def fit(self, dataset, n_epochs, batch_size):
                     X_w, X_d, y = X_w.cuda(), X_d.cuda(), y.cuda()
 
                 self.optimizer.zero_grad()
-                y_pred =  net(X_w, X_d)
-                loss = self.criterion(y_pred, y)
+                y_pred =  net(X_w, X_d) # [batch_size, 1]
+                loss = None
```
Comment (jrzaurin) [inline wide_deep/torch_model.py]: I don't think we need `loss = None`?

**Comment by jrzaurin** [comment]
Thanks! 

have a look to the couple of comments I made and once you have replied I will merge 

thanks again 😉

File: wide_deep/torch_model.py (hunk 1 above)
Comment (hyonaldo) [reply wide_deep/torch_model.py]: @jrzaurin 
- You're right.  I heard the functional backward code `F.sigmoid` was efficient in versions below 0.3.x.
- But in all later versions (>0.4.x, stable, etc.), `nn.functional.sigmoid` is deprecated.
https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1379
So I suggest using `torch.sigmoid` instead. :)

File: wide_deep/torch_model.py (hunk 2 above)
Comment (hyonaldo) [reply wide_deep/torch_model.py]: @jrzaurin 
This is just because of the following codes.
```
if(self.criterion == F.cross_entropy):
    loss = self.criterion(y_pred, y) #[batch_size, 1]
else:
    loss = self.criterion(y_pred, y.view(-1, 1)) #[batch_size, 1]
```
In the stable version of PyTorch,  the implicit dimension choice for {} has been deprecated when we use `F.softmax `.
https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1191
So, I think there are two different cases when we run `loss = self.criterion`.

Thank you very much for your review!! :)

Comment (jrzaurin) [reply wide_deep/torch_model.py]: is merged, but I meant precisely the `loss = None` line. 

There is no need of that line, right. The rest is fine

Comment (hyonaldo) [reply wide_deep/torch_model.py]: Now I can accurately understand what you mean. :)
You're right. Python doesn't need it.
I appreciate you merge my poor codes nevertheless you knew it.
